pkg/storage/tsdb/config.go

```diff
@@ -206,6 +212,9 @@ func (cfg *TSDBConfig) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.WALCompressionEnabled, "blocks-storage.tsdb.wal-compression-enabled", false, "True to enable TSDB WAL compression.")
 	f.IntVar(&cfg.WALSegmentSizeBytes, "blocks-storage.tsdb.wal-segment-size-bytes", wal.DefaultSegmentSize, "TSDB WAL segments files max size (bytes).")
 	f.BoolVar(&cfg.FlushBlocksOnShutdown, "blocks-storage.tsdb.flush-blocks-on-shutdown", false, "True to flush blocks to storage on shutdown. If false, incomplete blocks will be reused after restart.")
+	f.DurationVar(&cfg.CloseIdleTSDBInterval, "blocks-storage.tsdb.close-idle-tsdb-interval", 0, "How often to check for idle TSDB that can be closed. 0 means disabled.")
+	f.IntVar(&cfg.CloseIdleTSDBConcurrency, "blocks-storage.tsdb.close-idle-tsdb-concurrency", 10, "Limit the number of concurrently closing TSDB when idle.")
+	f.DurationVar(&cfg.CloseIdleTSDBTimeout, "blocks-storage.tsdb.close-idle-tsdb-timeout", 12*time.Hour, "If TSDB has not received any data for this duration, and all blocks from TSDB have been shipped, TSDB is closed and deleted from local disk. Only used if -blocks-storage.tsdb.close-idle-tsdb-interval is set.")
```

Comment:
Flushing is not enough... We don't want to close TSDBs too early to avoid wrong query results. This setting should be set to the same value (or higher) than `-querier.query-ingesters-within`.

Reply:
You're right. I would improve the comment about the relationship with `-querier.query-ingesters-within`.